Misc/NEWS.d/next/Library/2020-12-22-22-47-22.bpo-42721.I5Ai5L.rst

```diff
@@ -0,0 +1,8 @@
+When simple query dialogs (:mod:`tkinter.simpledialog`), message boxes
+(:mod:`tkinter.messagebox`) or color choose dialog
+(:mod:`tkinter.colorchooser`) are created without arguments *master* and
+*parent*, and the default root window is not yet created, a new temporary
+hidden root window will be created automatically. It will not be set as the
+default root window and will be destroyed right after closing the dialog
```

Comment:
The problem with this, to me, is that someone could be fooled into thinking that the temporary root will be created even if No_Default_Root has been called.  
"It will be set as the default root window only momentarily and will ..."

Reply:
Ignoring possible race conditions in multithread program (this is a separate issue), the default root window is not set for outer observer. The fact that it is actually set and immediately reset to None is an implementation detail.

But a note about NoDefaultRoot() will be added.